internal/flag/internal_flag_test.go

```diff
@@ -1710,6 +1711,111 @@ func TestInternalFlag_Value(t *testing.T) {
 	}
 }
 
+func TestInternalFlag_ValueWithBucketingKey(t *testing.T) {
+	type args struct {
+		flagName    string
+		user        ffcontext.Context
+		flagContext flag.Context
+	}
+	tests := []struct {
+		name  string
+		flag  flag.InternalFlag
+		args  args
+		want  interface{}
+		want1 flag.ResolutionDetails
+	}{
+		{
+			name: "Should use custom bucketing key when set",
+			flag: flag.InternalFlag{
+				Variations: &map[string]*interface{}{
+					"variation_A": testconvert.Interface("value_A"),
+					"variation_B": testconvert.Interface("value_B"),
+					"variation_C": testconvert.Interface("value_C"),
+				},
+				BucketingKey: "teamId",
+				Rules: &[]flag.Rule{
+					{
+						Query:           testconvert.String("key eq \"team-123\""),
+						VariationResult: testconvert.String("variation_B"),
+					},
+					{
+						Query:           testconvert.String("key eq \"user-key\""),
+						VariationResult: testconvert.String("variation_C"),
+					},
+				},
+				DefaultRule: &flag.Rule{
+					VariationResult: testconvert.String("variation_A"),
+				},
+				Metadata: &map[string]interface{}{
+					"teamId": "team-123",
+				},
+			},
+			args: args{
+				flagName: "my-flag",
+				user:     ffcontext.NewEvaluationContextBuilder("user-key").AddCustom("teamId", "team-123").Build(),
+				flagContext: flag.Context{
+					DefaultSdkValue: "value_default",
+				},
+			},
+			want: "value_B",
+			want1: flag.ResolutionDetails{
+				Variant:   "variation_B",
+				Reason:    flag.ReasonTargetingMatch,
+				RuleIndex: testconvert.Int(0),
+				Cacheable: true,
+				Metadata: map[string]interface{}{
+					"teamId": "team-123",
+				},
+			},
+		},
+		{
+			name: "Should not use user key when bucketing key is set",
+			flag: flag.InternalFlag{
+				Variations: &map[string]*interface{}{
+					"variation_A": testconvert.Interface("value_A"),
+					"variation_B": testconvert.Interface("value_B"),
+				},
+				BucketingKey: "region",
+				Rules: &[]flag.Rule{
+					{
+						Query:           testconvert.String("key eq \"user-key\""),
+						VariationResult: testconvert.String("variation_B"),
+					},
+				},
+				DefaultRule: &flag.Rule{
+					VariationResult: testconvert.String("variation_A"),
+				},
+				Metadata: &map[string]interface{}{
+					"region": "nl-nh",
+				},
+			},
+			args: args{
+				flagName: "my-flag",
+				user:     ffcontext.NewEvaluationContextBuilder("user-key").AddCustom("region", "nl-nh").Build(),
+				flagContext: flag.Context{
+					DefaultSdkValue: "value_default",
+				},
+			},
+			want: "value_A",
+			want1: flag.ResolutionDetails{
+				Variant:   "variation_A",
+				Reason:    flag.ReasonDefault,
+				Cacheable: true,
+				Metadata: map[string]interface{}{
+					"region": "nl-nh",
+				},
+			},
+		},
+	}
```

Comment:
> A good test here could be to have a percentage-based evaluation, set a targetingKey that affect to 1 variant and a teamId that affect to another variant.

I'm not sure what to do here

When I was overwriting the `key`, this was actually verifying that _any_ rules would be applied based on that. Now, if a BucketingKey is set, since the `key` remains the same, as far as I can tell there is no way to test that it is actually using the custom bucketing key.  Only that rules (like `teamId eq xxx`) see the attribute, but that's kind of orthogonal to the change.